commons/all/src/main/java/org/infinispan/commons/stat/SimpleTimerTracker.java

```diff
@@ -0,0 +1,43 @@
+package org.infinispan.commons.stat;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.DoubleAdder;
+import java.util.concurrent.atomic.LongAdder;
+
+import io.micrometer.core.instrument.util.TimeUtils;
+
+/**
+ * A simple implementation of {@link TimerTracker} that keep tracks of events and the sum of their duration.
+ *
+ * @since 15.1
+ */
+public class SimpleTimerTracker implements TimerTracker {
+   private final LongAdder counter;
+   private final DoubleAdder totalTime;
+
+   public SimpleTimerTracker() {
+      counter = new LongAdder();
+      totalTime = new DoubleAdder();
+   }
+
+   @Override
+   public void update(Duration duration) {
+      update(duration.toNanos(), TimeUnit.NANOSECONDS);
+   }
+
+   @Override
+   public void update(long value, TimeUnit timeUnit) {
+      counter.increment();
+      totalTime.add(TimeUtils.convert(value, timeUnit, TimeUnit.NANOSECONDS));
```

Comment:
is there some risk of overflow?

Reply:
eventually, the double max value is something like 1^1023 and the overflow would happen on the `sum()` method.